datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestStickyPartitionAssignment.java

```diff
@@ -415,6 +428,8 @@ public void testElasticTaskPartitionAssignmentCorrectlyAdjustsNumTasks() {
     // Partition assignment should go through
     assignment = strategy.assignPartitions(assignment, partitionsMetadata);
     validatePartitionAssignment(assignment, partitions, maxPartitionsPerTask, numTasksNeeded);
```

Comment:
Not sure if there is a direct way, but one test can be to have 1 instance, 1 task-A, 1 partition; Check the new metrics; then remove the task and add task with another prefix and see if the metric for the first prefix is no longer emitted and is emitted for the second prefix.

 since this is emitted only by the leader, but the metric is a gauge (which continuously emit), on zk expiry, the leader-> follower will continue to emit, while the new leader will also emit, resulting in incorrect value. So, if you introduce a deregisterMetrics path, you can validate that extra metrics are not emitted when not required i.e. no tasks for the datastream. (This is tricky and may need more thought, but since this is very leader specific, we either need to fix the gauge value somehow or deregister itself)

Reply:
We'd either need to extend MetricsAware or extend the AssignmentStrategy interface to add a method for deregistration of metrics. Is that okay with you? Also, depending on how this is done, will need to see how easy it is to simulate in tests.

Reply:
Don't need this test since we're not deregistering the metric, just resetting it. Which means the metric mbean will still exist, but have a value of 0.